sound/soc/sof/loader.c

```diff
@@ -129,6 +130,83 @@ int snd_sof_fw_parse_ext_data(struct snd_sof_dev *sdev, u32 bar, u32 offset)
 }
 EXPORT_SYMBOL(snd_sof_fw_parse_ext_data);
 
+static size_t snd_sof_ext_man_size(const void *fw)
```

Comment:
Yes, I can check file size before comparison, but I shouldn't return here error code when size will be smaller than `sizeof(*head)` because it's not a right job for this function to judge fw size. When size is smaller than `sizeof(*head)`  then we just know that there is not any ext_man, nothing more.